pkg/operator/starter.go

```diff
@@ -398,7 +398,7 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 	go connectivityCheckController.Run(ctx, 1)
 
 	<-ctx.Done()
-	return nil
+	return context.Cause(ctx)
```

Comment:
won't this report an error even when the ctx was cancelled ?
if yes, then do you know when the ctx can be cancelled ?
will the ctx be cancelled when the process receives a `SIGTERM` ?